crates/ty_python_semantic/src/types/tuple.rs

```diff
@@ -812,20 +853,66 @@ impl<'db> TupleSpec<'db> {
     }
 
     fn is_disjoint_from(&self, db: &'db dyn Db, other: &Self) -> bool {
+        // Two tuples with an incompatible number of required elements must always be disjoint.
+        if let ((minimum, _), (_, Some(maximum))) | ((_, Some(maximum)), (minimum, _)) =
+            (self.size_hint(), other.size_hint())
+        {
+            if maximum < minimum {
+                return true;
+            }
+        }
```

Comment:
I don't totally understand why this doesn't work when your previous `match` statement did (they seem like they should produce equivalent code!) -- but I think this should fix the regression caused by my previous suggestion, and it seems more obviously correct as well!

```suggestion
        let (self_min, self_max) = self.size_hint();
        let (other_min, other_max) = other.size_hint();
        if self_max.is_some_and(|max|max < other_min) {
            return true;
        }
        if other_max.is_some_and(|max|max < self_min) {
            return true;
        }
```